flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceSql.java

```diff
@@ -53,7 +55,11 @@ public class TestIcebergSourceSql extends TestSqlBase {
   @BeforeEach
   @Override
   public void before() throws IOException {
-    TableEnvironment tableEnvironment = getTableEnv();
+    setUpTableEnv(getTableEnv());
```

Comment:
Could you please help me understand why the streaming env is required? Is Flink watermark strategy restricted to streaming?

Reply:
Yes Source watermarks are being called on Source only in streaming env. 

For iceberg source even in Batch mode, it would still be important to configure a watermark column, so the splits are sorted. But applySourceWatermark is not being called in batch env.

More details here,
https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/dev/datastream/execution_mode/#event-time--watermarks

Reply:
Thx for clarifying